packages/wrangler/src/errors.ts

```diff
@@ -5,11 +5,26 @@
  * messaging.
  */
 export class UserError extends Error {
-	constructor(...args: ConstructorParameters<typeof Error>) {
-		super(...args);
+	cleanMessage: string | undefined;
+	constructor(
+		message?: string | undefined,
+		options?:
+			| (ErrorOptions & {
+					/**
+					 * This is used to provide telemetry with a sanitised error
+					 *  message that could not have any user-identifying information.
+					 * Use "*" as shorthand to duplicate `message`.
+					 *  */
+					cleanMessage?: string;
```

Comment:
Nit: How about `telemetryMessage`? I would prefer making it a bit more explicit for its purpose.

Also, instead of having a special shorthand, we can consider making this `boolean | string`. So that we can specify `true` if we are fine with the current message.